src/Symfony/Bridge/Doctrine/ContainerAwareEventManager.php

```diff
@@ -69,10 +69,9 @@ public function dispatchEvent($eventName, EventArgs $eventArgs = null): void
     public function getListeners($event = null): array
     {
         if (null === $event) {
-            trigger_deprecation('symfony/doctrine-bridge', '6.2', 'Calling "%s()" without an event name is deprecated. Call "getAllListeners()" instead.', __METHOD__);
-
-            return $this->getAllListeners();
+            throw new \InvalidArgumentException(sprintf('"%s()" must be called with an event name.', __METHOD__));
```

Comment:
The reason for deprecating `null` here was doctrine/event-manager#50. Simply throwing is not enough because as long as we declare compatibility with EventManager 1, we have to accept `null` as argument because that's the contract of EventManager 1.

What we need to do here is to declare the DoctrineBridge incompatible with `doctrine/event-manager` < 2. And once we did that, we can change the signature to…

```php
public function getListeners(string $event): array
```

Reply:
Updated dependencies as well as tests with the last deprecation present in `ContainerAwareEventManager` :+1: Thanks for the hints!